### PR TITLE
Add a chat provider agnostic way to determine a channel is not a direct message

### PR DESCRIPTION
**Pull request by dansmithy**
@cdupuis Support for the mapped param is in production (as well as an equivalent value in the source map). Please let me know if any further changes are required to get this in. Thanks.

### CODE REVIEW

File: lib/handlers/command/slack/AssociateRepo.ts [hunk 1]
```diff
@@ -129,6 +128,9 @@ export class AssociateRepo implements HandleCommand {
     @MappedParameter(MappedParameters.SlackChannelName)
     public channelName: string;
 
+    @MappedParameter("atomist://slack/channel_type")
```
Comment (cdupuis) [inline lib/handlers/command/slack/AssociateRepo.ts]: We probably want that to be a constant in automation-client for consistency. I can do that.

File: lib/handlers/command/slack/LinkRepo.ts [hunk 2]
```diff
@@ -115,7 +114,7 @@ export class LinkRepo implements HandleCommand {
             return ctx.messageClient.respond(err)
                 .then(() => Success, failure);
         }
-        if (!isChannel(this.channelId)) {
+        if (this.channelType !== "channel") {
```
Comment (cdupuis) [inline lib/handlers/command/slack/LinkRepo.ts]: What are possible values here? 

 * `channel`

**Review by cdupuis — COMMENTED** [review]
Couple of comments.

File: lib/handlers/command/slack/LinkRepo.ts (hunk 2 above)
Comment (dansmithy) [reply lib/handlers/command/slack/LinkRepo.ts]: `channel` or `user`. Would you rather I make an enum?

File: lib/handlers/command/slack/AssociateRepo.ts (hunk 1 above)
Comment (dansmithy) [reply lib/handlers/command/slack/AssociateRepo.ts]: Thanks!

**Review by cdupuis — CHANGES_REQUESTED** [review]
Just realised that this needs to be against the remove-index branch and not against master. Sorry.

I also released automation-client  2.0.0-master.20200311165330 what now has the new MappedParameters constant as well as an enum for the values.

**Comment by cdupuis** [comment]
@dansmithy where in the `source` structure is that new field? Do you have an example for that? I need that for the sdm command wrapping code.

**Comment by dansmithy** [comment]
> @dansmithy where in the `source` structure is that new field? Do you have an example for that? I need that for the sdm command wrapping code.

I've added a `:type` attribute to the `:channel` which has one of those two values.

Will post you an example.

**Review by cdupuis — APPROVED** [review]

**Pull request merged**